### PR TITLE
adds hotkey to toggle between walk and run

### LINKED ISSUE #2963: Change walk mode with hotkey
## Description
It should be possible to bind a hotkey to toggle/hold between walk and run mode.

### PR DESCRIPTION
**Pull request by corp-0**
# Pull Request Template

### Purpose
fix #2963
Adds a new bindable hotkey to toggle walk and run. Default key is C.
### Notes:
Players might need to reset their key settings in order to register the new hotkey in their systems.  At least I had to.

### Please make sure you have followed the self checks below before submitting a PR:

- Code is sufficiently commented
- Code is indented with tabs and not spaces
- The PR does not include any unnecessary .meta, .prefab or <b>.unity (scene) changes</b>
- The PR does not bring up any new compile errors
- The PR has been tested in editor
- Any new files are named using PascalCase (to avoid issues on case sensitive file systems)
- Any new / changed components follow the [Component Development Checklist](https://github.com/unitystation/unitystation/wiki/Component-Development-Checklist)
- Any new objects / items follow the [Creating Items and Objects Guide](https://github.com/unitystation/unitystation/wiki/Creating-Items-and-Objects%3A-Now-With-Prefab-Variants) (especially concerning the use of prefab variants)
- The PR has been tested in multiplayer (with 2 clients and late joiners, if applicable)
- The PR has been tested with round restarts.
- The PR has been tested on moving / rotating / rotated-before-joining matrices (if applicable)

### CODE REVIEW

**Comment by Jibbajabbafic** [comment]
Looks good to me. Would you be comfortable trying to add a way to detect missing keybinds and assign default keys if they don't exist yet? That would help with all future keybind changes. If not then I can make an issue for it.

**Comment by corp-0** [comment]
I'm not sure how to do that, but I could give it a try.

**Comment by corp-0** [comment]
@Jibbajabbafic done!

**Review by vlandemart — APPROVED** [review]

File: UnityProject/Assets/Scripts/Input System/KeybindManager.cs [hunk 1]
```diff
@@ -515,6 +518,31 @@ public void LoadKeybinds()
 				ResetKeybinds();
 				ModalPanelManager.Instance.Inform("Unable to read saved keybinds.\nThey were either corrupt or outdated, so they have been reset.");
 			}
+
+			
+			// Check if new hotkeys has been added to the default dict
```
Comment (Jibbajabbafic) [inline UnityProject/Assets/Scripts/Input System/KeybindManager.cs]: It's a good start but what if something gets added and something else gets removed?

If you use LINQ (stick `using System.Linq;` at the top) you can do something like this to find any dictionary keys that are different:
```
var keysDictionary1HasThat2DoesNot = dictionary1.Keys.Except(dictionary2.Keys);
var keysDictionary2HasThat1DoesNot = dictionary2.Keys.Except(dictionary1.Keys);
```
Then if there are any differences you can add/remove entries as necessary.

**Review by Jibbajabbafic — CHANGES_REQUESTED** [review]
Good stuff!
Just a slight change for some extra future proofing and it should be all good.

File: UnityProject/Assets/Scripts/Input System/KeybindManager.cs (hunk 1 above)
Comment (corp-0) [reply UnityProject/Assets/Scripts/Input System/KeybindManager.cs]: Nice, I will use that as soon as my local repo unfucks itself.

Comment (corp-0) [reply UnityProject/Assets/Scripts/Input System/KeybindManager.cs]: what type does this Except method returns? Is it just another dictionary?

Comment (corp-0) [reply UnityProject/Assets/Scripts/Input System/KeybindManager.cs]: Done. Works on my end, but tell me if that's what you asked.

**Review by Jibbajabbafic — COMMENTED** [review]
Code looks perfect now! I won't have a chance to test this in the editor till later though.

**Review by Jibbajabbafic — APPROVED** [review]
Works perfectly, thanks!

**Pull request merged**